persistence/commons/src/main/java/org/apache/isis/persistence/jpa/integration/changetracking/EntityChangeTrackerDefault.java

```diff
@@ -173,29 +148,98 @@ Set<PropertyChangeRecord> snapshotPropertyChangeRecords() {
         return entityPropertyChangeRecordsForPublishing.get();
     }
 
-    private boolean isEntityEnabledForChangePublishing(final @NonNull ManagedObject adapter) {
+    /**
+     * For any enlisted Object Properties collects those, that are meant for publishing,
+     * then clears enlisted objects.
+     */
+    private Set<PropertyChangeRecord> capturePostValuesAndDrain() {
```

Comment:
TODO: check whether this is in a thread-safe context otherwise needs synchronization